templates/java.lang.String.cpp

```diff
@@ -29,6 +36,16 @@ String& String::operator = (const String& other)
 	return *this;
 }
 
+String& String::operator = (String&& other)
+{
+	if (m_Str)
+		jni::ReleaseStringUTFChars(*this, m_Str);
```

Comment:
Don't you need to check if not assigning string to itself before doing this? I think we should check for that at the beginning of the method.